site/docs/sql/file-connector.md

```diff
@@ -0,0 +1,247 @@
+declaration, we infer names and types based on a sample.
+
+```sql
+CREATE MAPPING my_files
+TYPE File
+OPTIONS (
+    'path' = '/path/to/directory',
+    'format' = 'jsonl'
+)
+```
+
+#### Mapping Between JSON and SQL Types
+
+| JSON type | SQL Type  |
+| - | - |
+| `BOOLEAN` | `BOOLEAN` |
+| `NUMBER` | `DOUBLE` |
+| `STRING` | `VARCHAR` |
+| all other types | `OBJECT` |
+
+### Avro & Parquet Serialization
+
+The `avro` & `parquet` files are expected to contain Avro records.
+
+```sql
+CREATE MAPPING my_files
+TYPE File
+OPTIONS (
+    'path' = '/path/to/directory',
+    'format' = 'avro'
+)
+```
+
+```sql
+CREATE MAPPING my_files
+TYPE File
+OPTIONS (
+    'path' = 'hdfs://path/to/directory',
+    'format' = 'parquet'
+    /* more Hadoop options ... */
+)
+```
+
+#### Mapping Between Avro and SQL Types
+
+| Avro Type | SQL Type |
+| - | - |
+| `BOOLEAN` | `BOOLEAN` |
+| `INT` | `INT` |
+| `LONG` | `BIGINT` |
+| `FLOAT` | `REAL` |
+| `DOUBLE` | `DOUBLE` |
+| `STRING` | `VARCHAR` |
+| all other types | `OBJECT` |
+
+## External Name
+
+You rarely need to specify the columns in DDL. If you do, you might want
+to specify the external name. External names are supported for all
+formats except for `csv`.
+
+We don't support nested fields, hence the external name should refer to
+the top-level field - not containing any `.`.
+
+## File Table Functions
+
+To execute an ad hoc query against data in files you can use one of the
+predefined table functions:
+
+* `csv_file`
+* `jsonl_file`
+* `avro_file`
+* `parquet_file`
+
+Table functions will create a temporary mapping, valid for the duration
+of the statement. They accept the same options as `CREATE MAPPING`
+statements.
+
+You can use positional arguments:
+
+```sql
+SELECT * FROM TABLE(
+  CSV_FILE('/path/to/directory', '*.csv', MAP['key', 'value'])
+)
+```
+
+Or named arguments:
+
+```sql
+SELECT * FROM TABLE(
+  CSV_FILE(path => '/path/to/directory', options => MAP['key', 'value'])
+)
+```
+
+## Installation
+
+Depending on what formats you want to work with you need different
+modules on the classpath.
+
+`csv` format requires the `hazelcast-jet-csv` module:
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Gradle-->
+
+```groovy
+compile 'com.hazelcast.jet:hazelcast-jet-csv:{jet-version}'
+```
+
+<!--Maven-->
+
+```xml
+<dependency>
+    <groupId>com.hazelcast.jet</groupId>
+    <artifactId>hazelcast-jet-csv</artifactId>
+    <version>{jet-version}</version>
+</dependency>
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+`avro` format requires the `hazelcast-jet-avro` module:
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Gradle-->
+
+```groovy
+compile 'com.hazelcast.jet:hazelcast-jet-avro:{jet-version}'
+```
+
+<!--Maven-->
+
+```xml
+<dependency>
+    <groupId>com.hazelcast.jet</groupId>
+    <artifactId>hazelcast-jet-avro</artifactId>
+    <version>{jet-version}</version>
+</dependency>
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+`parquet` format and all remote files require the `hazelcast-jet-hadoop`
```

Comment:
This should be `hazelcast-jet-hadoop-all`.

Cloud storages need their respective jars, maybe just link to the section in sources and sinks.

Reply:
I have already linked file documentation from here to get rid of duplication.